src/test/scala/com/amazon/deequ/VerificationResultTest.scala

```diff
@@ -124,11 +124,11 @@ class VerificationResultTest extends WordSpec with Matchers with SparkContextSpe
         import session.implicits._
         val expected = Seq(
           ("group-1", "Error", "Success", "CompletenessConstraint(Completeness(att1,None))",
-            "Success", ""),
+            "Success", "Value: 1.0 does meet the constraint requirement!"),
           ("group-2-E", "Error", "Error", "SizeConstraint(Size(None))", "Failure",
             "Value: 4 does not meet the constraint requirement! Should be greater than 5!"),
           ("group-2-E", "Error", "Error", "CompletenessConstraint(Completeness(att2,None))",
-            "Success", ""),
+            "Success", "Value: 1.0 does meet the constraint requirement! Should equal 1!"),
           ("group-2-W", "Warning", "Warning",
```

Comment:
Can you clarify why the message contains the hint ("Should ...!") in one case, but not in the other?

Reply:
Sure, the reason is that the first check does not actually have a hint. In that case only the standard message is shown.

The checks are in the same test suite from line 201:

```scala
private[this] def getChecks(): Seq[Check] = {
    val checkToSucceed = Check(CheckLevel.Error, "group-1")
      .isComplete("att1")

    val checkToErrorOut = Check(CheckLevel.Error, "group-2-E")
      .hasSize(_ > 5, Some("Should be greater than 5!"))
      .hasCompleteness("att2", _ == 1.0, Some("Should equal 1!"))

    val checkToWarn = Check(CheckLevel.Warning, "group-2-W")
      .hasDistinctness(Seq("item"), _ < 0.8, Some("Should be smaller than 0.8!"))

    checkToSucceed :: checkToErrorOut :: checkToWarn :: Nil
}
```
  
The change allows the hint to be passed through even if the evaluation was a success, before that the hint was lost. Passing it through makes it a bit easier if the verification result is saved, because the hint can be used to know what the threshold at a given time was.